### DISCUSSION TITLE
Difference between service and flows

### DISCUSSION THREAD

**Discussion by robin5210** [Netflix/dispatch#5636] [Q&A] [answered]
I'm trying to understand the difference between the content of `flows.py` and `service.py` in a module.

It is not very clear for me how the correct file for a function is chosen. 

How I imaging it now it the following distinction:
**flow**: Processes some complex logic on the record. Doesn't generally returns data.
**service**: Retrieve records based on some given conditions. Doesn't contain much business logic.

Is this correct?

**Comment by mvilanova** [comment, marked as answer]
Hey @robin5210, that's correct. Service modules are primarily used for CRUD operations and flow modules for business logic. Here's an example, API request to create a new incident comes in, views module handles that and decides which flow within the flows module needs to be run, then the service module is used to create the incident in the db. Does that make sense?

**Reply by robin5210** [reply]
That totally makes sense. Thanks for the answer!
